src/MSBuild/CommandLineSwitches.cs

```diff
@@ -365,6 +372,43 @@ private struct ParameterizedSwitchInfo
             return false;
         }
 
+        /// <summary>
+        /// Get the distinct parameterized switchs map resource ids.
+        /// </summary>
+        /// <returns>Parameterized switches map resource ids.</returns>
+        internal static string[] GetParameterizedSwitchResourceIds()
+        {
+            List<string> parameterizedSwitchResourceIds = new List<string>();
+            for (int i = 0; i < (int)ParameterizedSwitch.NumberOfParameterizedSwitches; i++)
+            {
+                string resourceId = s_parameterizedSwitchesMap[i].resourceId;
+                if (!parameterizedSwitchResourceIds.Contains(resourceId) && !string.IsNullOrEmpty(resourceId))
+                {
+                    parameterizedSwitchResourceIds.Add(s_parameterizedSwitchesMap[i].resourceId);
+                }
+            }
+            return[.. parameterizedSwitchResourceIds];
```

Comment:
This looks like a great opportunity for `yield return`!
```suggestion
            foreach (ParameterizedSwitchInfo switch in s_parameterizedSwitchesMap)
            {
                if (!string.IsNullOrEmpty(switch.resourceId))
                {
                    yield return switch.resourceId;
                }
            }
```

Reply:
There are some switches using the same resource Id. So deal with duplicates here.  But thanks for your suggestions. Have refactored the code.